internal/controller/configuration_monitor.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntime "sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/stolostron/siteconfig/internal/controller/configuration"
+)
+
+const (
+	// SiteConfigOperatorConfigMap defines the name of the ConfigMap used to manage
+	// the SiteConfig Operator's runtime configuration.
+	SiteConfigOperatorConfigMap = "siteconfig-operator-configuration"
+
+	// RequeueDelay defines the default requeue delay on transient errors.
+	RequeueDelay = 30 * time.Second
+)
+
+// ConfigurationMonitor watches for changes to a specific ConfigMap and updates
+// the SiteConfig Operator's runtime configuration accordingly.
+type ConfigurationMonitor struct {
+	client.Client
+	Log    *zap.Logger
+	Scheme *runtime.Scheme
+
+	// Namespace containing the ConfigMap
+	Namespace string
+
+	// Store for runtime configuration
+	ConfigStore *configuration.ConfigurationStore
+}
+
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile processes events for the SiteConfig Configuration ConfigMap, ensuring the runtime configuration
+// remains synchronized.
+// Creates a default Configuration ConfigMap if one does not exist.
+func (r *ConfigurationMonitor) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	log := r.Log.With(
+		zap.String("name", req.Name),
+		zap.String("namespace", req.Namespace),
+	)
+
+	log.Info("Start reconcile")
+	defer log.Info("Completed reconcile")
+
+	// Ignore non-SiteConfig configuration ConfigMaps
+	if req.Name != SiteConfigOperatorConfigMap {
```

Comment:
Shouldn't this be handled by the filter? Although I suppose it's fine to have a safety check here :shrug:

Reply:
In theory it should be handled by the filter function. However, it's an added, low-cost safety check.